x-pack/plugins/lens/public/xy_visualization/xy_config_panel/color_picker.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './xy_config_panel.scss';
+import React, { useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
+import { EuiFormRow, EuiColorPicker, EuiColorPickerProps, EuiToolTip, EuiIcon } from '@elastic/eui';
+import type { PaletteRegistry } from 'src/plugins/charts/public';
+import type { VisualizationDimensionEditorProps } from '../../types';
+import { State } from '../types';
+import { FormatFactory } from '../../../common';
+import { getSeriesColor } from '../state_helpers';
+import { getAccessorColorConfig, getColorAssignments } from '../color_assignment';
+import { getSortedAccessors } from '../to_expression';
+import { updateLayer } from '.';
+
+const tooltipContent = {
+  auto: i18n.translate('xpack.lens.configPanel.color.tooltip.auto', {
+    defaultMessage: 'Lens automatically picks colors for you unless you specify a custom color.',
+  }),
+  custom: i18n.translate('xpack.lens.configPanel.color.tooltip.custom', {
+    defaultMessage: 'Clear the custom color to return to “Auto” mode.',
+  }),
+  disabled: i18n.translate('xpack.lens.configPanel.color.tooltip.disabled', {
+    defaultMessage:
+      'Individual series cannot be custom colored when the layer includes a “Break down by.“',
+  }),
+};
+
+export const ColorPicker = ({
+  state,
+  setState,
+  layerId,
+  accessor,
+  frame,
+  formatFactory,
+  paletteService,
+}: VisualizationDimensionEditorProps<State> & {
+  formatFactory: FormatFactory;
+  paletteService: PaletteRegistry;
+}) => {
+  const index = state.layers.findIndex((l) => l.layerId === layerId);
+  const layer = state.layers[index];
+  const disabled = !!layer.splitAccessor;
```

Comment:
```suggestion
  const disabled = Boolean(layer.splitAccessor);
```
I don't mind `!!` but I heard once we prefer sticking to converting via`Boolean` ;)